pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -80,6 +84,21 @@ contract ManagedPool is ManagedPoolSettings {
         // solhint-disable-previous-line no-empty-blocks
     }
 
+    // Virtual Supply
+
+    /**
+     * @notice Returns the number of tokens in circulation.
+     * @dev In other pools, this would be the same as `totalSupply`, but since this pool pre-mints BPT and holds it in
+     * the Vault as a token, we need to subtract the Vault's balance to get the total "circulating supply". Both the
+     * totalSupply and Vault balance can change. If users join or exit using swaps, some of the preminted BPT are
+     * exchanged, so the Vault's balance increases after joins and decreases after exits. If users call the recovery
+     * mode exit function, the totalSupply can change as BPT are burned.
+     */
+    function _getVirtualSupply() internal view override returns (uint256) {
+        (uint256 cash, uint256 managed, , ) = getVault().getPoolTokenInfo(getPoolId(), IERC20(this));
+        return totalSupply() - (cash + managed);
```

Comment:
omg y u no safemath

`cash + managed` is fine because of vault restrictions (we should mention that), but there's no guarantees on `totalSupply`.

Reply:
We're subtracting here so the Vault would have to hold more BPT than there is in existence for this to underflow.

Reply:
Why do you always have to be right